httpx/_client.py

```diff
@@ -493,6 +493,9 @@ def _redirect_stream(
         return request.stream
 
 
+_Client = typing.TypeVar("_Client", bound="Client")
```

Comment:
Wondering if we should have a comment here linking to https://www.python.org/dev/peps/pep-0484/#annotating-instance-and-class-methods (?)

Reply:
Did, and added some reasons behind the change here 😉

Reply:
What about just naming it `ClientType`?

Reply:
Or maybe that would imply `Type[Client]`..

Reply:
May just be bikeshedding, if so ignore me. I'm not sure if there is actually a name that neatly encapsulates the concept of an upper bounded type variable. Maybe not worth trying :)

Reply:
Possible colors to paint the shed: `TClient` `AnyClient`

Reply:
I guess(?) I'd probably prefer that we follow the PEP and stick with `T`.

Reply:
So, what's the conclusion here? I can make a change

Reply:
I'd also personally go for `T` (or `C`, or whatevs), as `_Client` feels more like a "private base class" than a "generic bounded type variable".

Reply:
Will make a change. Keep in mind that switching to `T = TypeVar('T', covariant=True)` loses the ability to use `TypeVar(.., bound=...)`. After all, `T` is a generic name used everywhere and python's typing system does not support namespace.

Reply:
changes pushed

Reply:
Oops, without `bound=...`, mypy complains

Reply:
As I predicted, lacking of namespace causes trouble. The following works:
```python
T = typing.TypeVar("T", bound="Client")
U = typing.TypeVar("U", bound="AsyncClient")
```
What do you think?